File: intellecta-backend/src/main/java/com/intellecta/intellecta_backend/service/NotesService.java
package com.intellecta.intellecta_backend.service;

import com.intellecta.intellecta_backend.dto.request.NoteRequest;
import com.intellecta.intellecta_backend.dto.response.NoteResponse;
import java.util.List;

public interface NotesService {
    NoteResponse createNote(Long userId, NoteRequest request);
    List<NoteResponse> getAllNotes(Long userId);
    NoteResponse getNoteById(Long userId, Long noteId);
    NoteResponse updateNote(Long userId, Long noteId, NoteRequest request);
    void deleteNote(Long userId, Long noteId);
    List<NoteResponse> searchNotes(Long userId, String q, String tag);
    NoteResponse flagForReview(Long userId, Long noteId);
    NoteResponse togglePin(Long userId, Long noteId);
    List<NoteResponse> getReviewQueue(Long userId);
}

File: intellecta-frontend/src/services/notesService.js
import api from "./api";

import { getUserId } from "../utils/auth";

const uid = () => {
  const id = getUserId();
  if (!id) throw new Error("User not authenticated");
  return id;
};

export const getAllNotes = () => api.get(`/notes/user/${uid()}`);

export const createNote = (noteData) =>
  api.post(`/notes/user/${uid()}`, noteData);

export const updateNote = (noteId, noteData) =>
  api.put(`/notes/user/${uid()}/${noteId}`, noteData);

export const deleteNote = (noteId) =>
  api.delete(`/notes/user/${uid()}/${noteId}`);

export const searchNotes = (q = "", tag = "") =>
  api.get(`/notes/user/${uid()}/search`, { params: { q, tag } });

export const togglePin = (noteId) =>
  api.patch(`/notes/user/${uid()}/${noteId}/pin`);

export const flagForReview = (noteId) =>
  api.patch(`/notes/user/${uid()}/${noteId}/review`);

export const getReviewQueue = () =>
  api.get(`/notes/user/${uid()}/review-queue`);
